File: week1/Erenimo3442/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        low = 0
        high = len(numbers) - 1
        while True:
            middle = (low + high) // 2
            sum = numbers[low] + numbers[high]
            if sum == target:
                return [low + 1, high + 1]
            elif sum < target:
                if numbers[high] + numbers[middle] < target:
                    low = middle + 1
                else:
                    low += 1
            elif sum > target:
                if numbers[low] + numbers[middle] > target:
                    high = middle - 1
                else:
                    high -= 1

File: week1/SalihSamedDinc/two-sum-ii-input-array-is-sorted.c
vector<int> twoSum(vector<int>& numbers, int target) {
    int low = 0;
    int high = numbers.size() - 1;
    while(low <= high) {
        if(numbers[low] + numbers[high] == target) return {low + 1, high + 1};
        else if(numbers[low] + numbers[high] < target) {
            low++;
        } else {
            high--;
        }
    }
    return {};
}